packages/calling-stateful-client/review/beta/calling-stateful-client.api.md

```diff
@@ -191,9 +194,14 @@ export interface RemoteVideoStreamState {
     view?: VideoStreamRendererViewState;
 }
 
+// @public
+export interface StatefulCallAgentOptions extends CallAgentOptions {
+    alternativeCallerId?: string;
+}
+
 // @public
 export interface StatefulCallClient extends CallClient {
-    createCallAgent(...args: Parameters<CallClient['createCallAgent']>): Promise<DeclarativeCallAgent>;
+    createCallAgent(tokenCredential: CommunicationTokenCredential, options?: StatefulCallAgentOptions): Promise<DeclarativeCallAgent>;
```

Comment:
It makes sense to me that Contoso needs to provide `alternativeCallerId` when creating the adapter -- that is the only place Contoso has to provide this information.

It also makes sense to me that we need to save `alternativeCallerId` *somewhere* in the `CallClientState`, because it is needed by the handlers for the `StartCallButton`.. The alternative would be for Contoso to provide the `alternativeCallerId` directly to the props for `StartCallButton`, but that would break the nice property that they simply need to pass in `usePropsFor(StartCallButton)` to get the button working.


That said, I'm uncomfortable with this extension of `CallAgentOptions` and `CallAgentState`.

The important distinction between `alternativeCallerId` and most of the fields in `CallClientState` is that we never recieve `alternativeCallerId` from the ACS services -- this is not a cached value we get from the backed. In that way, this is similar to `CallClientState.userId`.
I'd also expect `CallClientState.userId` to only be used from handlers coresponding to `CallAgent` methods, similar to the handlers that use `alternativeCallerId`.

I'd prefer to keep our departure from headless SDK APIs in one place -- while creating the `StatefulCallClient`, and store `alternativeCallerId` alongside `userId`.

I think it's worth bringing this up in the team channel to see what folks think.

Reply:
So this phone number that is going in `alternativeCallerId` is from the ACS backend. it is required that your phone number comes from the same connection string and resource that you are running your app with. That said I have been going back and forth with where to store this phone number, both places (callClientState & callAgentState) I think have good reasons to have it. I will post something about this in the teams chat.